plugin/externalPluginEntry.go

```diff
@@ -372,25 +372,28 @@ func (e *externalPluginEntry) Metadata(ctx context.Context) (JSONObject, error)
 }
 
 func (e *externalPluginEntry) Stream(ctx context.Context) (io.ReadCloser, error) {
-	inv := e.script.NewInvocation(ctx, "stream", e)
+	cancelCtx, cancelFunc := context.WithCancel(ctx)
```

Comment:
Hmm, why not call cancelFunc before closing the stream in https://github.com/puppetlabs/wash/blob/master/cmd/validate.go#L329 ?

Reply:
I suppose that would work. But the semantics of closing the stream would still be wrong.

Reply:
What are the semantics of closing the stream? I was thinking something like "Close is only called when the context is cancelled."

Reply:
io.Closer has its own definition. It feels weird to say you should only call this function after the context is cancelled. But it would be a smaller change for sure.

Reply:
Ah ok, so this is to make sure that stdoutStreamer properly implements io.Closer semantics. That makes sense. If that's the case, maybe an `internal.Command#Terminate/Kill` helper could be useful? It's more expressive of what you want to do when closing the stream, but also a bit more invasive.

Reply:
I think if you choose the `cancelCtx` approach, a comment explaining why it's used would help (something like "We use cancelCtx to ensure that stdoutStreamer kills the process if it is closed for some other reason besides context cancellation").